pyproject.toml

```diff
@@ -138,40 +141,35 @@ extend-select = [
     "RET",         # flake8-return
     "RSE",         # flake8-raise
     "RUF",         # Ruff-specific
+#    "S",           # flake8-bandit
     "SLF",         # flake8-self
     "SLOT",        # flake8-slots
     "SIM",         # flake8-simplify
+#    "T20",         # flake8-print
     "TCH",         # flake8-type-checking
     "TID",         # flake8-tidy-imports
     "TRY",         # tryceratops
     "UP",          # pyupgrade
     "YTT",         # flake8-2020
 ]
 extend-ignore = [
-    "PLR",  # Design related pylint codes
-    "ISC001", # Added because of ruff-format warning
+    "ANN101",  # Missing type annotation for self in method
+    "ANN102",  # Missing type annotation for cls in classmethod
+    "ISC001",  # Conflicts with formatter
+    "E501",    # Line too long (Black is enough)
+    "PLR",     # Design related pylint codes
+    "S101",    # Use of assert detected
 ]
+flake8-unused-arguments.ignore-variadic-names = true
 isort.required-imports = ["from __future__ import annotations"]
 
-[tool.ruff.per-file-ignores]
+[tool.ruff.lint.per-file-ignores]
 "*.pyi" = ["D"]  # pydocstyle
 "*.ipynb" = [
     "D",    # pydocstyle
     "E402", # Allow imports to appear anywhere in Jupyter notebooks
     "I002", # Allow missing `from __future__ import annotations` import
 ]
 
-[tool.ruff.pydocstyle]
+[tool.ruff.lint.pydocstyle]
 convention = "google"
-
-
-[tool.pytest.ini_options]
-filterwarnings = [
-    "ignore:Call to deprecated create function FieldDescriptor",
-    "ignore:Call to deprecated create function Descriptor",
-    "ignore:Call to deprecated create function EnumDescriptor",
-    "ignore:Call to deprecated create function EnumValueDescriptor",
-    "ignore:Call to deprecated create function FileDescriptor",
-    "ignore:Call to deprecated create function OneofDescriptor",
-    "ignore:distutils Version classes are deprecated. Use packaging.version instead.",
-]
```

Comment:
I'd advise you to copy over the configuration from MQT Bench here.
A proper pytest configuration is actually part of the scientific Python repo review checklist.
Should be easy and fast to adapt from the latest mqt bench update.

Reply:
I adjusted it according to the MQT Bench config.